### ISSUE TITLE
PHP 8 Incompatibility with sending reset password emails.

### ISSUE THREAD

**Issue by Monica-Wood** [collectiveaccess/providence#1385]
I have enabled auth_allow_password_reset and tried to send a password reset email.  
on branch /dev/php8

I am getting the below stacktrace

`Fatal error: Uncaught TypeError: array_key_exists(): Argument #2 ($array) must be of type array, Zend_Registry given in /var/www/ca/vendor/magento/zendframework1/library/Zend/Registry.php:207 Stack trace: #0 /var/www/ca/vendor/magento/zendframework1/library/Zend/Registry.php(183): Zend_Registry->offsetExists() #1 /var/www/ca/vendor/magento/zendframework1/library/Zend/Validate/Abstract.php(419): Zend_Registry::isRegistered() #2 /var/www/ca/vendor/magento/zendframework1/library/Zend/Validate/Abstract.php(376): Zend_Validate_Abstract::getDefaultTranslator() #3 /var/www/ca/vendor/magento/zendframework1/library/Zend/Validate/Abstract.php(218): Zend_Validate_Abstract->getTranslator() #4 /var/www/ca/vendor/magento/zendframework1/library/Zend/Validate/Abstract.php(294): Zend_Validate_Abstract->_createMessage() #5 /var/www/ca/vendor/magento/zendframework1/library/Zend/Validate/Hostname.php(1860): Zend_Validate_Abstract->_error() #6 /var/www/ca/vendor/magento/zendframework1/library/Zend/Validate.php(98): Zend_Validate_Hostname->isValid() #7 /var/www/ca/vendor/magento/zendframework1/library/Zend/Mail/Protocol/Smtp.php(190): Zend_Validate->isValid() #8 /var/www/ca/vendor/magento/zendframework1/library/Zend/Mail/Transport/Smtp.php(200): Zend_Mail_Protocol_Smtp->helo() #9 /var/www/ca/vendor/magento/zendframework1/library/Zend/Mail/Transport/Abstract.php(348): Zend_Mail_Transport_Smtp->_sendMail() #10 /var/www/ca/vendor/magento/zendframework1/library/Zend/Mail.php(1201): Zend_Mail_Transport_Abstract->send() #11 /var/www/ca/app/helpers/mailHelpers.php(192): Zend_Mail->send() #12 /var/www/ca/app/helpers/mailHelpers.php(282): caSendmail() #13 /var/www/ca/app/models/ca_users.php(3075): caSendMessageUsingView() #14 /var/www/ca/app/models/ca_users.php(2971): ca_users->sendPasswordResetMail() #15 /var/www/ca/app/controllers/system/AuthController.php(123): ca_users->requestPasswordReset() #16 /var/www/ca/app/lib/Controller/RequestDispatcher.php(284): AuthController->RequestPassword() #17 /var/www/ca/app/lib/Controller/AppController.php(114): RequestDispatcher->dispatch() #18 /var/www/ca/index.php(119): AppController->dispatch() #19 {main} thrown in /var/www/ca/vendor/magento/zendframework1/library/Zend/Registry.php on line 207`

**Comment by Monica-Wood** [comment]
I have determined that the issue is in the upstream vendor library magento/zendframework1 and have temporarily fixed my issue and reported the bug to upstream: https://github.com/magento/zf1/issues/54

**Comment by collectiveaccess** [comment]
Updated mailer functions to use PHPMailer as there was no movement with Magento to fix this issue. 41c8b624f74f7d956fc2b7f413d736252cadb5a2

**Comment by Monica-Wood** [comment]
Thanks for working on this.
I tested it today and discovered it didn't work for me.  Looking through the code I notice you are forcing SSL traffic to use port 587. I understand this is the standard port for SSL, but not all mail providers are using this yet and are still using port 465 (as mine is).  Can I request that you use the port that people enter into the settings instead of overriding it?

[https://github.com/collectiveaccess/providence/blob/dev/php8/app/helpers/mailHelpers.php#L136](url)
I updated this to  `$o_mail->Port = $smtp_config['port'];` and my mail send worked straight away.
